demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -32,18 +33,24 @@
 
 import static com.novoda.downloadmanager.DownloadBatchStatus.Status.ERROR;
 
+// Need to extract collaborators for this demo to reduce complexity. GH Issue #286
+@SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.StdCyclomaticComplexity", "PMD.ModifiedCyclomaticComplexity"})
```

Comment:
This needs to be dealt with before release but would make this PR way larger and obscure the important parts. GH issue here #286